RealmSwift/Sync.swift

```diff
@@ -545,3 +567,479 @@ extension Realm {
+                } else {
+                    promise(.success(updateResult))
+                }
+            }
+        }
+    }
+    
+    /// Overwrites a single document in a collection based on a query filter and
+    /// returns the document in either its pre-replacement or post-replacement
+    /// form. Unlike `updateOneDocument`, this action allows you to atomically find,
+    /// replace, and return a document with the same command. This avoids the
+    /// risk of other update operations changing the document between separate
+    /// find and update operations.
+    /// - Parameters:
+    ///   - filter: A `Document` that should match the query.
+    ///   - replacement: A `Document` describing the replacement.
+    ///   - options: `RLMFindOneAndModifyOptions` to use when executing the command.
+    func findOneAndReplace(filter: Document, replacement: Document) -> Future<[String: RLMBSON]?, Error> {
+        return Future { promise in
+            self.findOneAndReplace(filter: filter, replacement: replacement) { updateResult, error in
+                if let error = error {
+                    promise(.failure(error))
+                } else {
+                    promise(.success(updateResult))
+                }
+            }
+        }
+    }
+    
+    /// Removes a single document from a collection based on a query filter and
+    /// returns a document with the same form as the document immediately before
+    /// it was deleted. Unlike `deleteOneDocument`, this action allows you to atomically
+    /// find and delete a document with the same command. This avoids the risk of
+    /// other update operations changing the document between separate find and
+    /// delete operations.
+    /// - Parameters:
+    ///   - filter: A `Document` that should match the query.
+    ///   - options: `FindOneAndModifyOptions` to use when executing the command.
+    func findOneAndDelete(filter: Document, options: FindOneAndModifyOptions) -> Future<[String: RLMBSON]?, Error> {
+        return Future { promise in
+            self.findOneAndDelete(filter: filter, options: options) { deleteResult, error in
+                if let error = error {
+                    promise(.failure(error))
+                } else {
+                    promise(.success(deleteResult))
+                }
+            }
+        }
+    }
+    
+    /// Removes a single document from a collection based on a query filter and
+    /// returns a document with the same form as the document immediately before
+    /// it was deleted. Unlike `deleteOneDocument`, this action allows you to atomically
+    /// find and delete a document with the same command. This avoids the risk of
+    /// other update operations changing the document between separate find and
+    /// delete operations.
+    /// - Parameters:
+    ///   - filter: A `Document` that should match the query.
+    func findOneAndDelete(filter: Document) -> Future<[String: RLMBSON]?, Error> {
```

Comment:
All of these methods need to return `AnyBSON` instead of `RLMBSON`. There is a support method in the ObjectiveC Support file that can convert this.

Reply:
Also AnyBSON should be optional